index.js

```diff
@@ -1091,27 +1123,56 @@
           this._registerTrackers();
         }
 
-        // activates page tracking
-        if (trackRoutes) {
-          $rootScope.$on(pageEvent, function () {
-            // Apply $route based filtering if configured
-            if (readFromRoute) {
+        // This section adds different listeners based on the configuration
+        var eventListener = null;
+        if (typeof trackRouteFn === 'function') {
+          eventListener = function() {
+            // Redirect event to custom function, passing in protected object
+            trackRouteFn.apply(that, arguments);
+          };
+        } else if (trackRoutes) {
+          if (hybridMobileSupport) {
+            // For mobile devices we need to track a screen rather than a page
+            eventListener = function (event, state) {
+              if(!mobileAppProperties || !mobileAppProperties.appName) {
+                $log.warn('Analytics can not track mobile apps without basic \'appName\' properties');
+                return;
+              }
+              // Create a copy for the tracking function to avoid manipulation during transmission
+              var args = angular.copy(mobileAppProperties);
+              args.screenName = state.name;
```

Comment:
We may need to differentiate between `ngRoute` and `ui-router` here. The sample taken from #145 seems to use `ui-router` while our default is `ngRoute`.